### ISSUE TITLE
3DS redirection page does not return back to the app without a user click

### ISSUE THREAD

**Issue by ayi-smartwave** [braintree/braintree_android#198]
### General information

* Braintree version: 2.10.0, 2.11.0, and 2.12.0
* Payment method: Card nonce payment method with ThreeDSecure check
* Environment: Release
* Build types: Both debug and release
* Android Version and Device: 
    - Asus Zenfone Max 5.5 with Android 6.0.1
    - Samsung A7 with Android 5.0.2
    - Lenovo TAB 2 A7-30HC with Android 5.0.1
    - Nexus S API 26 (Emulator) with Android 8.0
    - Pixel API 26 (Emulator) with Android 8.0
    - Pixel 2 API 26 (Emulator) with Android 8.0

### Issue description

Repro steps:
* Add item to cart
* Make sure there's a saved payment
* Logout
* Login account
* Go to checkout
* Click pay

Result: The phone's browser app will be opened to this page

![screenshot_20180410-143126_20180410_143137118](https://user-images.githubusercontent.com/24887050/38545865-a5ed8c22-3cdd-11e8-8266-ddcb1adb3303.jpg)

then after a few seconds, it will be redirected to the app.

Expected Result: Payment will continue as normal

Note that this only happens if the user just logged in. If you try clicking pay again, payment will continue as normal. In short this is unwanted browser redirection.

**Comment by quinnjn** [comment]
Hi @ayi-smartwave,

Can you let us know what payment method you're debugging? 
Is this happening in both Braintree sandbox and release environments, or are those build types?

**Comment by ayi-smartwave** [comment]
Card nonce payment method with ThreeDSecure check. Those are build types. We encounter it on Braintree release environment.

**Comment by ayi-smartwave** [comment]
I'd like a follow up on this, thanks.

**Comment by quinnjn** [comment]
So I think you're asking why this page exists after the 3DS flow but before returning to the app.

The current 3DS flow isn't ideal and due to Chrome security concerns the user may not be redirected back into the app. This extra step exists for that case. If Chrome does not redirect the customer back into the app, the customer can click on the button to register an interaction that will allow Chrome to redirect.

For example: In Android P you should notice that it doesn't redirect at all and you have to click on the button to continue.

We're going to clean up the wording for this redirect page, but we're open for suggestions on a better solution for this!

**Comment by ayi-smartwave** [comment]
Okay thank you.

**Comment by wedjohn57** [comment]
I am facing the issue as well. I need to know, by default will Chrome automatically close? For my case, Chrome will never close at all no matter what phone model or OS.

**Comment by sdcoffey** [comment]
Hey @wedjohn57,

Can you elaborate on your question? Like @quinnjn said above, for versions of Android below API 28, Chrome should automatically redirect. Is there anything in your logs to indicate some other error?

**Comment by quinnjn** [comment]
We updated the redirect page wording for the 3DS redirect page in 3.4.1. 

I'm going to close this issue with these facts:

- We fixed the wording to ask the customer to click the redirect button to go back to the app.
- For Android P+: Chrome security concerns do not allow us to automatically redirect. Customers must click the redirect button to go back to the app.
- We don't have the ability to tell Chrome to close. The tab will exist harmlessly until the customer closes the tab themselves.
